### PR TITLE
[1LP][RFR] Collections For Container Objects

### PR DESCRIPTION
**Pull request by jawatts**
The scope of this PR is to create Collection Classes for all of the Container Objects. The all method for each container collection object will be handled in a seperate PR 

@gshefer will be updating the tests in a separate PR

Manual Testing Below (Few examples):
```
In [17]: ocp = providers.get_crud('cm-juwatts')

In [18]: appliance = store.current_appliance

In [19]: coll_containers = appliance.collections.containers

In [20]: coll_containers
Out[20]: ContainerCollection(filters={})

In [21]: view_containers = navigate_to(coll_containers, 'All')

In [22]: view
Out[22]: <cfme.containers.container.ContainerAllView at 0x7f6f08de4a90>

In [23]: coll_image_reg = appliance.collections.container_image_registries

In [24]: coll_image_reg
Out[24]: ImageRegistryCollection(filters={})

In [25]: view_image_reg = navigate_to(coll_image_reg, 'All')

In [26]: view_image_reg
Out[26]: <cfme.containers.image_registry.ImageRegistryAllView at 0x7f6ef3f65d10>

In [28]: coll_pods = appliance.collections.container_pods

In [29]: coll_pods
Out[29]: PodCollection(filters={})

In [30]: view_pods = navigate_to(coll_pods, 'All')

In [31]: view_pods
Out[31]: <cfme.containers.pod.PodAllView at 0x7f6ef3f65c50>


```

shriver:
{{ pytest: cfme/tests/containers/test_cockpit.py cfme/tests/containers/test_node.py --use-provider cm-env2 --long-running }}

### CODE REVIEW

File: cfme/containers/volume.py [hunk 1]
```diff
@@ -45,7 +45,26 @@ def get_random_instances(cls, provider, count=1, appliance=None):
...
+    """Collection object for :py:class:`Volume`."""
+
+    ENTITY = Volume
+
+    def all(self, provider):
+        # container_volumes table has ems_id, join with ext_mgmgt_systems on id for provider name
+        volume_table = self.appliance.db.client['container_volumes']
+        ems_table = self.appliance.db.client['ext_management_systems']
+        volume_query = self.appliance.db.client.session.query(volume_table.name)\
+            .join(ems_table, volume_table.parent_id == ems_table.id)
```
Comment (jawatts) [inline cfme/containers/volume.py]: Notice it is parent_id here not ems_id to link to the provider

File: cfme/containers/image.py [hunk 2]
```diff
@@ -140,9 +139,8 @@ def all(self):
             ems_table, image_table.ems_id == ems_table.id)
         images = []
 
-        for name, id, provider_name in image_query.all():
-            images.append(self.instantiate(name=name, id=id,
-                provider=ContainersProvider(name=provider_name, appliance=self.appliance)))
+        for name, id in image_query.all():
+            images.append(self.instantiate(name=name, id=id, provider=provider))
```
Comment (jawatts) [inline cfme/containers/image.py]: @gshefer Note here: I changed this from instantiating a generic containers provider to the provider under test

File: setup.py [hunk 3]
```diff
@@ -93,7 +93,7 @@
             'projects = cfme.configure.access_control:ProjectCollection',
             'candus = cfme.configure.configuration.region_settings:CANDUCollection',
             'groups = cfme.configure.access_control:GroupCollection',
-            'nodes = cfme.containers.node:NodeCollection',
+            'container_nodes = cfme.containers.node:NodeCollection',
```
Comment (jawatts) [inline setup.py]: I updated nodes to container_nodes to keep consistency, tests have been updated accordingly.

File: cfme/containers/node.py [hunk 4]
```diff
@@ -104,14 +104,13 @@ def all(self):
         nodes = []
         for name, provider_name in node_query.all():
             # Hopefully we can get by with just provider name?
-            nodes.append(self.instantiate(name=name,
-                                          provider=ContainersProvider(name=provider_name,
-                                                                      appliance=self.appliance)))
+            nodes.append(
+                self.instantiate(
+                    name=name, provider=ContainersProvider(
+                        name=provider_name, appliance=self.appliance)))
```
Comment (mshriver) [inline cfme/containers/node.py]: name passed twice

File: cfme/containers/image.py [hunk 5]
```diff
@@ -129,21 +128,8 @@ def compliant(self):
 @attr.s
 class ImageCollection(BaseCollection):
     """Collection object for :py:class:`Image`."""
-    ENTITY = Image
-
-    def all(self):
```
Comment (mshriver) [inline cfme/containers/image.py]: Add this back in, will modify in future PRs

Comment (jawatts) [reply cfme/containers/image.py]: Done

File: cfme/tests/containers/test_node.py [hunk 6]
```diff
@@ -22,7 +22,7 @@ def test_nodes_navigate(soft_assert, appliance):
             test_item = appliance.collections.nodes
```
Comment (mshriver) [inline cfme/tests/containers/test_node.py]: this needs a `container_nodes` update

Comment (jawatts) [reply cfme/tests/containers/test_node.py]: Fixed, thought I caught them all.

**Comment by mshriver** [comment]
The CMQE team has other PRs that will be using these new collections/entity objects. I'm going to merge this so that the other PRs can be worked/reviewed this sprint.

**Pull request merged**